examples/client.rs

```diff
@@ -175,7 +177,7 @@ async fn main() -> Result<(), BQError> {
 
     client
         .tabledata()
-        .insert_all(project_id, dataset_id, table_id, insert_request)
+        .insert_all(project_id, dataset_id, table_id, insert_request, false)
```

Comment:
I would like to avoid this type of public API change when feasible. IMO, the right way to do it will be to change the `ServiceAccountAuthenticator` (in auth.rs) to support workload identity in addition to the other methods. This structure is initialized by the `Client` via one of the constructors currently available. You could introduce a new constructor `Client::with_workload_identity()` to initialize properly the ServiceAccountAuthenticator.